next-app/BUILD.bazel

```diff
@@ -0,0 +1,61 @@
+load("@npm//:next/package_json.bzl", next_bin = "bin")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("@aspect_rules_js//js:defs.bzl", "js_binary", "js_run_binary", "js_test")
+
+npm_link_all_packages(name = "node_modules")
+
+filegroup(
+    name = "srcs",
+    srcs = glob([
+        "pages/**",
+        "public/**",
+        "styles/**",
+    ]) + [
+        ".eslintrc.json",
+        "next-env.d.ts",
+        "next.config.js",
+        "package.json",
+        "tsconfig.json",
+    ],
+)
+
+# This custom next js_binary is needed since next is very sensitive to it being found in two
+# `node_modules` trees. With the standard generated `bin.next` build rule loaded from
+# `@npm//:next/package_json.bzl`, next is found both in the binary's runfiles `node_modules` and in
+# the execroot's `:node_modules` tree. This breaks the build. The work-around here is to have this
+# `next_bin` `js_binary` be an empty shim without any `node_modules` and re-direct from it to run
+# next from its execroot `:node_modules/next/dist/bin/next` entry point.
```

Comment:
fascinating, I guess we should have an issue to document this on rules_js somewhere as well?